tests/cuda_ndarrays/conftest.py

```diff
@@ -0,0 +1,132 @@
+# pylint: disable=missing-function-docstring, missing-module-docstring/
+# pylint: disable=arguments-differ, inconsistent-return-statements, protected-access, abstract-method/
+import subprocess
+import os
+import pathlib
+import sys
+import shutil
+import pytest
+
+NEEDS_FROM_PARENT = hasattr(pytest.Item, "from_parent")
+
+def pytest_collect_file(parent, path):
+    """
+    A hook to collect test_*.cu test files.
+
+    """
+    if path.ext == ".cu" and path.basename.startswith("test_"):
+        if NEEDS_FROM_PARENT:
+            return CTestFile.from_parent(path=pathlib.Path(path), parent=parent)
+        return CTestFile(parent=parent, path=pathlib.Path(path))
+
+def pytest_collection_modifyitems(items):
+    """
+    a hook to modify the items before the tests for C Cuda unit test files.
+
+    """
+    for item in items:
+        if item.fspath.ext == ".cu":
+            item._nodeid = item.nodeid + " < " + item.test_result["DSCR"] + " >"
+
+class CTestFile(pytest.File):
+    """
+    A custom file handler class for C unit test files.
+
+    """
+
+    @classmethod
+    def from_parent(cls, **kwargs):
+        return super().from_parent(**kwargs)
+
+    def collect(self):
+        """
+        Overridden collect method to collect the results from each
+        C unit test executable.
+
+        """
+        # Run the exe that corresponds to the .c file and capture the output.
+        test_exe = os.path.splitext(str(self.fspath))[0]

```

Comment:
What is the type of `self.fspath`? I'm surprised that it needs to be converted to a string. I would have expected it to either already be a string, or be a type which has functions for handling this kind of operation

Reply:
Same question for `self.config.rootdir` on the next line

Reply:
`self.fspath` is an attribute of type `py._path.local.localPath` which why it needs to be converted to a string. As an alternative I can use the attribute `basename` to get the filename with the extension as a string without the need for conversion.

Reply:
Same for `self.config.rootdir`. It does have also its own way of handling the conversion  which is the attribute `strpath`.

Reply:
Updated here 4fd629acda3bc1e8540ead05ed2162c4e8c3530a